### PR TITLE
Vladimir.makotkin table part

### PR DESCRIPTION
**Pull request by notvoveex**
Add:
 - Row class implementation / test coverage;
 - Table class header;

Update:
 - Variant class methods(const modifier for getters);

### CODE REVIEW

File: src/Table.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    void setConfig(const std::map<std::string, DataTypes>& col_cfg);
+
+    void addRow(const Row& row);
+    void setRow(const Row& row, std::function<void()>);  // `std::function<...>` signature hasn't been defined yet
+    void removeRow(std::function<void()>);
+
+    void setTableName(const std::string& new_table_name) { table_name_ = new_table_name; }
+    const std::string& getTableName() { return table_name_; }
+
+    bool getBoolItem(std::function<void()>);  // `std::function<...>` signature hasn't been defined yet
```
Comment (lucky-rydar) [inline src/Table.h]: You dont need getbool getint etc.. because you have variant class

File: src/Table.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    int getIntItem(std::function<void()>);
+    float getFloatItem(std::function<void()>);
+    const std::string& getStringItem(std::function<void()>);
+
+    void setItem(bool data, std::function<void()>);  // `std::function<...>` signature hasn't been defined yet
```
Comment (lucky-rydar) [inline src/Table.h]: Here also add only set varian

**Review by lucky-rydar — CHANGES_REQUESTED** [review]
Some fixes.

Also add searching cell in table by column name

File: src/Table.cpp [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+}
+
+Table::Table(const std::string& table_name, const config_t& table_config, const table_t& sortedTables)
+        : table_name_(table_name), table_config_(table_config), table_(sortedTables)
+{
+    // pass
+}
+
+
+void Table::setTName(const std::string &new_table_name)
```
Comment (lucky-rydar) [inline src/Table.cpp]: Why it is everywhere 'T' in the middle of name of function?

**Review by lucky-rydar — COMMENTED** [review]
There some letters in functions names of Table class.
The code looks pretty good, i hope you have tested it and it works correctly.
As for select method, later it will he rewrote to run some query written from console

**Comment by notvoveex** [comment]
> Some fixes.
> 
> Aslo add searching cell in table by column name

Done.

File: src/Table.cpp (hunk 3 above)
Comment (notvoveex) [reply src/Table.cpp]: > Why it is everywhere 'T' in the middle of name of function?

Done.

**Pull request merged**